test/orderController.test.js

```
@@ -0,0 +1,132 @@
+    };
+  });
+
+  //create an order -Success case
+  it('should create an order successfully', async () => {
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+    const mockPaymentIntent = {
+      id: 'pi_mock123',
+      client_secret: 'mock_secret',
+    };
+    //data for order
+    let orderData = {
+      orderItems: [
+        {
+          album: album._id,
+          quantity: 2,
+        },
+      ],
+      subtotal: 100,
+      tax: 10,
+      total: 110,
+      user: user._id,
+      paymentIntentId: mockPaymentIntent.id,
+    };
+
+    //AKOS: this one gives 500 status code
+    response = await request(app)
+      .post(`/api/v1/orders`)
+      .set('Cookie', signedCookie)
+      .send(orderData);
+
+    expect(response.status).toBe(StatusCodes.CREATED);
+    expect(response.body).toHaveProperty(
+      'clientSecret',
+      mockPaymentIntent.client_secret
+    );
+    expect(response.body).toHaveProperty('order');
+  });
+
+  //create an order -Error case- user tries to create an order without logging in
+  it('should create an order successfully- Error case- user was not logged in', async () => {
+    //data for order
+    const orderData = {
+      orderItems: [
+        {
+          album: album._id,
+          quantity: 2,
+        },
+      ],
+      subtotal: 100,
+      tax: 10,
+      total: 110,
+      user: user._id,
+    };
+
+    response = await request(app).post(`/api/v1/orders`).send(orderData);
+    expect(response.status).toBe(StatusCodes.UNAUTHORIZED);
+  });
+  //create an order -Error case - some data is missing
+  //get 401 instead of 400.. not sure why
+  it('should return a 400 status if order items are missing', async () => {
+    const orderData = {
+      orderItems: [],
+      subtotal: 100,
+      tax: 10,
+      total: 110,
+      user: user._id,
+    };
+
+    const response = await request(app).post('/api/v1/orders').send(orderData);
+
+    expect(response.status).toBe(StatusCodes.BAD_REQUEST);
```

This test right now is doing a great job of catching an issue in our server code :star\_struck: 

Here in the test we're expecting a `400` bad request response, which would make sense for what to return if the user input is missing some expected values. But instead, the endpoint returns a `500`. This is happening because instead of letting our `error-handler` middleware handle errors that occur in this endpoint, we've wrapped all the code in the controller in a `try/catch` (needed to do our transaction correctly) and then for _any_ error caught, we're manually returning a 500: [https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/testOrdersEndpoints/src/controllers/orderController.js#L75](https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/testOrdersEndpoints/src/controllers/orderController.js#L75)

The suggested fix here would be to update the `orderController.js` code to re-raise the error (`throw error;` ) instead of of returning a `res.status(500)` in the `catch` block. Then, that means that the `error-handler` middleware will get to handle the error as usual and it can decide based on the type of error, what status code and message to return.